modules/sharedaddy/recaptcha.php

```diff
@@ -128,11 +128,23 @@ public function verify( $response, $remote_ip ) {
 		if ( true !== $resp_decoded['success'] ) {
 			return new WP_Error( $error_code, $error_message );
 		}
-
 		// Validate the hostname matches expected source
 		if ( isset( $resp_decoded['hostname'] ) ) {
 			$url = wp_parse_url( get_home_url() );
-			if ( $url['host'] !== $resp_decoded['hostname'] ) {
+
+			/**
+			 * Allow other valid hostnames.
+			 *
+			 * This can be useful in cases where the token hostname is expected to be
+			 * different from the get_home_url (ex. AMP recaptcha token contains a different hostname)
+			 *
+			 * @since 9.1.0
+			 *
+			 * @param array [ $url['host'] ] List of the valid hostnames to check against.
+			 */
+			$valid_hostnames = apply_filters( 'jetpack_recaptcha_valid_hostnames', [ $url['host'] ] );
```

Comment:
While we can use short array syntax in PHP 5.6+, WordPress core opted against recommending their use in their standards. I would consequently recommend that we follow Core's lead. You can [find out more here](https://make.wordpress.org/core/2019/07/12/php-coding-standards-changes/).

```suggestion
			$valid_hostnames = apply_filters( 'jetpack_recaptcha_valid_hostnames', array( $url['host'] ) );
```